lib/resque/worker.rb

```diff
@@ -144,11 +151,14 @@ def work(interval = 5.0, &block)
 
           done_working
           @child = nil
+
+          run_hook :after_perform, self
         else
-          break if interval.zero?
-          log! "Sleeping for #{interval} seconds"
-          procline paused? ? "Paused" : "Waiting for #{@queues.join(',')}"
-          sleep interval
+          break if interval.zero? # for testing
+          if paused?
+            procline "Paused"
+            sleep interval
```

Comment:
TODO: investigate the motivation for this change; moves the `sleep` to only happen if `paused?` instead of also happening if `reserve` returns `nil`.

Reply:
Likely related to the nonblocking pop?